### DISCUSSION TITLE
Сортировка по рейтингу

### DISCUSSION THREAD

**Discussion by NovomirK** [petrovvlad/freeLib#73] [General]
Добрый вечер. Не могу понять принцип сортировки. Открываю, к примеру, фантастику. Хочу отсортировать по рейтингу. Чтобы книги с 5 звёздами, соответственно были сверху. А не получается. Какая-то логика сортировки есть, но понять я её не могу.

И, (может быть как следствие?) когда в настройках конвертации в mobi ставишь галочку чтобы все книги серии были объеденены в один файл, последний том обычно идёт первым. 
Вот такие вот интересные аномалии замечены.

**Comment by petrovvlad** [comment]
Внёс изменения в сортировку книг. Пока только в develop ветку. Посмотрите. 
Код для загрузки develop ветки:
`git clone --recurse-submodules -b develop https://github.com/petrovvlad/freeLib.git`

Сортировку при конвертации ещё не смотрел.

**Comment by NovomirK** [comment]
Установил. Сейчас получается сделать сортировку по рейтингу по одной серии книг. А , к примеру, отсортировать лучшее, или по дате добавления в Жанре, не получается.

**Reply by petrovvlad** [reply]
На данный момент список книги представлен только в виде дерева. Сортировка возможна только внутри ветки "Автор - Серия". Для сортировки по всему списку надо реализовать вывод в виде таблицы.

**Reply by NovomirK** [reply]
Понятно. Может быть, когда-нибудь Вам будет скучно жить ... )
С объединением всё оказалось проще. Он вообще не объединяет файлы. Он их сохраняет по одному и иногда разбрасывает по разным файлам. Это я по недосмотру подумал что он какой-то том ставитт первым.

**Reply by petrovvlad** [reply]
Мне пока не скучно жить. Но вывод списка книг в виде таблицы сделал ;).

**Reply by NovomirK** [reply]
Здорово! Спасибо большое! Всё работает. 
 С объединением странно. Одну серию он объеденил. Остальные не хочет. Логики в действиях не нахожу. Но это уже совсем незначительно. Ещё раз спасибо за Ваш труд.

**Reply by petrovvlad** [reply]
Исправил объединение книг в один файл. Сортировку книг внутри файла пока не смотрел. Проверьте пожалйста.

**Reply by NovomirK** [reply]
Спасибо огромное. К сожалению не смогу до среды. Уехал в гости на длинные выходные. Как только вернусь, первым делом проверю.

**Reply by NovomirK** [reply]
Проверил. Всё работает, включая сортировку книг. Спасибо огромное! По-моему сейчас это идеальная библиотека.

**Reply by NovomirK** [reply]
И как только полностью настроена mobi амазон заявляет об окончании их поддержки в августе. ))) Буде теперь тестировать epub.

**Reply by NovomirK** [reply]
С объединением epub такая же история как была с mobi

**Reply by petrovvlad** [reply]
Придётся подождать. Я в отпуске :)

**Reply by NovomirK** [reply]
Хорошего отдыха!

**Reply by NovomirK** [reply]
не буду новую тему создавать :) Вот что нашёл: В табличном режиме ничего не конвертится и не отправляется.

**Reply by petrovvlad** [reply]
Я вернулся :)
Экспорт в табличном режиме поправил, пока в develop ветке. Проверяйте.
Не стесняйтесь создавать отдельные темы для новых ошибок. Только не в discussions, а в issues.

**Reply by NovomirK** [reply]
С возвращением! :) Всё заработало. В том числе и объединение серии в одну книгу в Epub. Чтобы при объединении правильно расставляло книги (по порядку) нужно предварительно отсортировать их по №. 
С темами понял, прошу прощения, исправлюсь :)
